compiler/ir/ir.objcinterop/src/org/jetbrains/kotlin/ir/objcinterop/ObjCInterop.kt

```diff
@@ -266,9 +267,14 @@ fun IrClass.getExternalObjCClassBinaryName(): String =
         this.getExplicitExternalObjCClassBinaryName()
                 ?: this.name.asString()
 
-fun IrClass.getExternalObjCMetaClassBinaryName(): String =
-        this.getExplicitExternalObjCClassBinaryName()
-                ?: this.name.asString().removeSuffix("Meta")
+fun IrClass.getExternalObjCMetaClassBinaryName(): String {
+    this.getExplicitExternalObjCClassBinaryName()?.let { return it }
+    // External ObjC metaclass is named as `Foo.Componion`, Foo is subclass of NSObject or itself.
+    if (this.isCompanion) {
+        (this.parent as? IrLazyClass)?.let { return it.name.asString() }
```

Comment:
`IrLazyClass` is an implementation detail. It should be enough to cast to `IrClass` instead.

Reply:
Fixed